prow/tide/tide.go

```diff
@@ -1410,15 +1422,19 @@ func (c *Controller) dividePool(pool map[string]PullRequest, pjs []prowapi.ProwJ
 		}
 		sps[fn].prs = append(sps[fn].prs, pr)
 	}
-	for _, pj := range pjs {
-		if pj.Spec.Type != prowapi.PresubmitJob && pj.Spec.Type != prowapi.BatchJob {
-			continue
-		}
-		fn := poolKey(pj.Spec.Refs.Org, pj.Spec.Refs.Repo, pj.Spec.Refs.BaseRef)
-		if sps[fn] == nil || pj.Spec.Refs.BaseSHA != sps[fn].sha {
-			continue
+
+	for subpoolkey, sp := range sps {
+		pjs := &prowapi.ProwJobList{}
+		err := c.prowJobClient.List(
+			c.ctx,
+			pjs,
+			ctrlruntimeclient.MatchingField(cacheIndexName, cacheIndexKey(sp.org, sp.repo, sp.branch, sp.sha)),
+			ctrlruntimeclient.InNamespace(c.config().ProwJobNamespace))
+		if err != nil {
+			return nil, fmt.Errorf("failed to list jobs for subpool %s: %v", subpoolkey, err)
 		}
-		sps[fn].pjs = append(sps[fn].pjs, pj)
+		c.logger.WithField("subpool", subpoolkey).Infof("Found %d prowjobs.", len(pjs.Items))
```

Comment:
nit: `Debugf`. Also might want to mention that this is the number of prowjobs found for the subpool _before_ filtering.

Reply:
Which filtering are you referring to by "before filtering"? They are already filtered by org/repo/branch+baseSHA